src/java/org/tensorics/core/tensorbacked/TensorbackedInternals.java

```diff
@@ -1,108 +1,157 @@
+     */
+    public static <V, T extends Tensorbacked<V>> T createBackedByTensor(Class<T> tensorBackedClass, Tensor<V> tensor) {
+        Tensor<V> dimensionAdjustedTensor = ensureSameDimensions(tensorBackedClass, tensor);
+        if (tensorBackedClass.isInterface()) {
+            return ProxiedInterfaceTensorbackeds.create(tensorBackedClass, dimensionAdjustedTensor);
+        } else {
+            return createByConstructor(tensorBackedClass, dimensionAdjustedTensor);
+        }
+    }
+
+    private static <V, T extends Tensorbacked<V>> T createByConstructor(Class<T> tensorBackedClass, Tensor<V> tensor) {
+        return instantiatorFor(tensorBackedClass).ofType(CONSTRUCTOR).withArgumentType(Tensor.class).create(tensor);
+    }
+
+    static <T extends Tensorbacked<V>, V> void verifyDimensions(Class<T> tensorBackedClass, Tensor<V> tensor) {
+        Set<Class<?>> targetDimensions = dimensionsOf(tensorBackedClass);
+        Set<Class<?>> tensorDimensions = tensor.shape().dimensionSet();
+
+        if (!Positions.areDimensionsConsistent(targetDimensions, tensorDimensions)) {
+            throw new IllegalArgumentException("Dimensions of tensorbacked target class (" + targetDimensions
+                    + ") do not match dimensions of given tensor (" + tensorDimensions + "). Cannot create tensorbacked object.");
+        }
+    }
+
+    public static <S> Iterable<Tensor<S>> tensorsOf(Iterable<? extends Tensorbacked<S>> tensorbackeds) {
+        List<Tensor<S>> tensors = new ArrayList<>();
+        for (Tensorbacked<S> tensorbacked : tensorbackeds) {
+            tensors.add(tensorbacked.tensor());
+        }
+        return tensors;
+    }
+
+    public static final <TB extends Tensorbacked<?>> Iterable<Shape> shapesOf(Iterable<TB> tensorbackeds) {
+        return Streams.stream(tensorbackeds).map(tb -> tb.tensor().shape()).collect(toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    public static final <TB extends Tensorbacked<?>> Class<TB> classOf(TB tensorBacked) {
+        return (Class<TB>) tensorBacked.getClass();
+    }
+
+    /*
+     * This was extracted from the abstract tensorbacked, as it was implemented there by michi. To be consisten this is
+     * now applied to all kind of instantiation techniques.
+     *
+     * However, there are 2 questions:
+     *
+     * * What is the original reason to potentially mutate the dimensions of the tensor here? Couldnt we be more strict
+     * and require the equality of the dimensionset? Anyhow, the builders should enforce this....or am I (KF)
+     * overlooking something?
+     *  * For the moment the same calls are kept in the abstract tensorbacked... However, they could potentially be removed,
+     *  except we assume that somebody calls this constructor directly...?
+     */
+    public static <TB extends Tensorbacked<E>, E> Tensor<E> ensureSameDimensions(Class<TB> tensorbackedClass, Tensor<E> tensor) {
```

Comment:
If I remember correctly this is to properly treat inheritance in dimensions, since we allow interfaces there.

Consider interface I and class A implementing it. Then you have:
Ti = Tensorbacked<Double> with dimension I
Ta = Tensorbacked<Double> with dimension A
in this case you should be able to do DoubleTensorics.calculate(Ti).plus(Ta) which should return a Tensorbacked in dimension I.

Reply:
Ok, I think this makes sense. I will put this into the javadoc.

Reply:
I changed the comment a bit, although I am not sure at the moment if this would not work otherwise ... but indeed it seems to be a good approach to align the dimensions.